### PR TITLE
Add CodeQL scan github action

### LINKED ISSUE #1287: Integrate CodeQL analysis
**Describe the feature request**
Integrate CodeQL scan via a github action to run at a regular interval.
From https://codeql.github.com/
```
Discover vulnerabilities across a codebase with CodeQL,
our industry-leading semantic code analysis engine.
CodeQL lets you query code as though it were data.
Write a query to find all variants of a vulnerability, eradicating it forever.
Then share your query to help others do the same.

CodeQL is free for research and open source.
```
also available as github action: https://github.com/github/codeql-action

**Describe alternatives you've considered**
Dependabot, which is already integrated with the repo. CodeQL provides additional analysis and surfaces any probable improvements we can make in the code.
 
**Affected product area (please put an X in all that apply)**

- [ ] APIs
- [ ] Addons
- [ ] CLI
- [ ] Docs
- [ ] IAM
- [ ] Installation
- [ ] Plugin
- [x] Security
- [x] Test and Release
- [ ] User Experience

**Additional context**

### PR DESCRIPTION
**Pull request by navidshaikh**
### What this PR does / why we need it
- Add CodeQL action github workflow to scan
     main and release branches.
- The results are uploaded in security tab of the repo for
     further investigation and fixes.
- Scan for languages: go and javascript
- Schedule: Everyday at 1 AM UTC

 Signed-off-by: Navid Shaikh <navids@vmware.com>

### Which issue(s) this PR fixes

Fixes #1287 

### Describe testing done for PR

Github action run on fork: https://github.com/navidshaikh/tanzu-framework/actions/workflows/security_codeql.yaml
Results: https://github.com/navidshaikh/tanzu-framework/security/code-scanning

### Release note

```release-note
Add CodeQL scan github action
```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

File: .github/workflows/security_codeql.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+name: 'Security - CodeQL scan (periodic)'
+
+on:
+  schedule:
+    - cron: '* 1 * * *'
+  pull_request:
+    branches: [ 'main', 'release-*' ]
+  push:
+    branches: [ 'pr/codeql' ]
```
Comment (navidshaikh) [inline .github/workflows/security_codeql.yaml:9]: this to debug/show the workflow runs on PR branch

Comment (navidshaikh) [reply .github/workflows/security_codeql.yaml]: removed the debug commit

File: .github/workflows/security_codeql.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+name: 'Security - CodeQL scan (periodic)'
+
+on:
+  schedule:
+    - cron: '* 1 * * *'
```
Comment (navidshaikh) [inline .github/workflows/security_codeql.yaml:5]: (now) will run periodically

File: .github/workflows/security_codeql.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    - name: Get dependencies
+      run: |
+        go get -v -t -d ./...
+
+    # Initializes the CodeQL tools for scanning.
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v1
+      with:
+        languages: go, javascript
```
Comment (navidshaikh) [inline .github/workflows/security_codeql.yaml:35]: configured for go and javascript

**Review by navidshaikh — COMMENTED** [review]
@vuil ready for review

**Review by vuil — APPROVED** [review]
lgtm, thanks

**Pull request merged**